### DISCUSSION TITLE
Planetscale CLI - Handshake error

### DISCUSSION THREAD

**Discussion by litcodeur** [planetscale/discussion#316] [Bug Reports] [answered]
Hello, I'm trying to connect to planetscale locally via the CLI and I'm receiving an error message stating "couldn't initial TLS handshake to remote addr: x509 <pscale_url> certificate is expired" when I try to connect to planetscale on my local system at the url `mysql://127.0.0.1/<db_name>` either via prisma on nodejs or via [tableplus](https://tableplus.com/). 

Interestingly when I run the `pscale connect <db_name> <db_branch>` command it says that secure connection is established, the error only shows when I actually try to connect to DB in any app.

```
pscale connect <db_name> <db_branch> --port 3309
Secure connection to database <db_name> and branch <db_branch> is established!.

Local address to connect your application: 127.0.0.1:3309 (press ctrl-c to quit)

//--- This error happens when I try to connect ---//
2022-10-22T16:17:24-04:00	error	error proxying conns	{"error": "couldn't initiate TLS handshake to remote addr: x509: “*.us-east-3.psdb.cloud” certificate is expired"}
```

Here's the version of cli and the system I'm running into this issue on
1. CLI version: pscale version 0.116.0 (build date: 2022-09-13T23:31:20Z commit: 4b2ef1e)
2. OS: MacOS Monterey 12.5.1

**Comment by servoice** [comment]
I am also having the same issue on AWS, vercel, and locally. 

`Error opening a TLS connection: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1921: (certificate has expired)`

**Comment by teaguestockwell** [comment]
I can repro this as well:


## vercel function
```
[GET] /api/get-presigned-pic-upload
13:28:50:57
2022-10-22T20:28:50.663Z	
ERROR	PrismaClientInitializationError: 
Invalid `prisma.rateLimit.count()` invocation:
Error opening a TLS connection: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1921: (certificate has expired)
    at RequestHandler.handleRequestError (/var/task/node_modules/@prisma/client/runtime/index.js:29760:13)
    at RequestHandler.request (/var/task/node_modules/@prisma/client/runtime/index.js:29737:12)
    at async PrismaClient._request (/var/task/node_modules/@prisma/client/runtime/index.js:30709:16)
    at async /var/task/.next/server/chunks/197.js:520:19
    at async /var/task/.next/server/chunks/7243.js:87:9 {
  clientVersion: '4.3.0',
  errorCode: undefined
}
```

## vercel build
```
> Build error occurred
--
13:42:46.422 | Error:
13:42:46.423 | Invalid `prisma.post.findMany()` invocation:
13:42:46.423 |  
13:42:46.423 |  
13:42:46.424 | Error opening a TLS connection: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1921: (certificate has expired)
13:42:46.424 | at RequestHandler.handleRequestError (/vercel/path0/node_modules/@prisma/client/runtime/index.js:29760:13)
13:42:46.428 | at RequestHandler.request (/vercel/path0/node_modules/@prisma/client/runtime/index.js:29737:12)
13:42:46.429 | at async PrismaClient._request (/vercel/path0/node_modules/@prisma/client/runtime/index.js:30709:16)
13:42:46.429 | at async getStaticPaths (/vercel/path0/.next/server/pages/r/[topicName]/post/view/[postId].js:2038:19)
13:42:46.429 | at async buildStaticPaths (/vercel/path0/node_modules/next/dist/build/utils.js:487:31)
13:42:46.429 | at async /vercel/path0/node_modules/next/dist/build/utils.js:622:115
13:42:46.429 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20) {
13:42:46.429 | type: 'PrismaClientInitializationError',
13:42:46.430 | clientVersion: '4.3.0'
13:42:46.430 | }
13:42:46.482 | error Command failed with exit code 1.
13:42:46.483 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
13:42:46.508 | Error: Command "yarn run build" exited with 1
```

## local prisma client

```
Message: Error in Prisma Client request: 
Invalid `prisma.storePart.findMany()` invocation:
Error opening a TLS connection: The certificate was not trusted.
```

**Reply by focux** [reply]
I'm having this issue as well, looks like something is wrong on PS end.

**Comment by teaguestockwell** [comment]
This looks like it is being tracked here:

https://www.planetscalestatus.com/incidents/m9bksfmfbmhx

**Comment by nickvanw** [comment, marked as answer]
Hey Folks,

My sincere apologies from the PlanetScale side - we had a cluster that did not refresh its TLS certificates, which caused an expired certificate to be offered for new connections, causing the issue in this discussion. 

We have remediated the problem and will be adding additional monitoring to make sure this does not happen in the future. This is far from the experience that we want our PlanetScale should expect, and we will be undergoing internal retrospectives next week to address it.

Have a great weekend.
